### PR TITLE
refactor: migrate trace storage to CH

### PR DESCRIPTION
**Pull request by erenming**
#### What this PR does / why we need it:
migrate trace storage to Clickhouse

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda-org.erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJhc3NpZ25lZSI6WyIxMDAwMzAxIl0sImxhYmVsIjpbMTQ1NV19&id=307200&iterationID=1218&pId=0&type=REQUIREMENT)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |       migrate trace storage to Clickhouse       |
| 🇨🇳 中文    |        trace数据存储迁移为Clickhouse      |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/oap/collector/core/model/exporter.go [hunk 1]
```diff
@@ -16,59 +16,99 @@ package model
...
+
+	Buffer           *odata.Buffer
+	Timer            *time.Timer
+	Interval, Jitter time.Duration
+}
+
+func (re *RuntimeExporter) Add(od odata.ObservableData) {
+	re.Buffer.Push(od)
+
+	if re.Buffer.Full() {
```
Comment (snakorse) [inline modules/oap/collector/core/model/exporter.go:44]: consider thread safety

File: modules/oap/collector/plugins/exporters/clickhouse/span/writer.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+					metas[i].SeriesID = sid
+					err := metaBatch.AppendStruct(&metas[i])
+					if err != nil {
+						ws.logger.Errorf("metaBatch append: %s", err)
+						_ = metaBatch.Abort()
+						putMetaBuf(metas)
+						goto begin
+					}
+				}
+				putMetaBuf(metas)
```
Comment (snakorse) [inline modules/oap/collector/plugins/exporters/clickhouse/span/writer.go:106]: `putMetaBuf `should outside the `if` branch

File: modules/oap/collector/core/model/exporter.go (hunk 1 above)
Comment (erenming) [reply modules/oap/collector/core/model/exporter.go:44]: There is a mutex in Buffer, so this is thread safe

File: cmd/monitor/streaming/bootstrap.yaml [hunk 3]
```diff
@@ -521,6 +521,64 @@ notify-storage:
...
+  pipelines:
+    spans:
+      - receivers:
+          - "erda.oap.collector.receiver.kafka@spotspan"
+          - "erda.oap.collector.receiver.kafka@oapspan"
+        processors:
+          - "erda.oap.collector.processor.modifier@span"
+        exporters:
+          - "erda.oap.collector.exporter.clickhouse@span"
+        batch_size: 1000
```
Comment (snakorse) [inline cmd/monitor/streaming/bootstrap.yaml:535]: make batch_size configurable by env

File: cmd/monitor/streaming/bootstrap.yaml [hunk 4]
```diff
@@ -521,6 +521,64 @@ notify-storage:
...
+    - action: drop
+      key: tags.http_request_content_length
+    - action: drop
+      key: tags.http_wrote_bytes
+# ************* processors *************
+
+# ************* exporters *************
+erda.oap.collector.exporter.clickhouse@span:
+  _enable: ${WRITE_SPAN_TO_CLICKHOUSE_ENABLE:false}
+  currency_num: ${EXPORTER_CLICKHOUSE_CURRENCY_NUM:30}
```
Comment (snakorse) [inline cmd/monitor/streaming/bootstrap.yaml:577]: 30 may be too large for the clickhouse, as clickhouse expect less concurrency writer with large size batch. use 3 as default

File: cmd/monitor/streaming/bootstrap.yaml [hunk 5]
```diff
@@ -521,6 +521,64 @@ notify-storage:
...
+# ************* processors *************
+erda.oap.collector.processor.modifier@span:
+  _enable: ${WRITE_SPAN_TO_CLICKHOUSE_ENABLE:false}
+  rules:
+    - action: drop
+      key: tags.http_read_bytes
+    - action: drop
+      key: tags.http_request_content_length
+    - action: drop
+      key: tags.http_wrote_bytes
```
Comment (snakorse) [inline cmd/monitor/streaming/bootstrap.yaml:571]: how to do with high cardinality tags which can not be dropped

File: cmd/monitor/streaming/conf/clickhouse/spans/ddl_create_default_tables.sql.tpl [hunk 6]
```diff
@@ -0,0 +1,29 @@
+CREATE TABLE IF NOT EXISTS <database>.spans_series ON CLUSTER '{cluster}'
```
Comment (snakorse) [inline cmd/monitor/streaming/conf/clickhouse/spans/ddl_create_default_tables.sql.tpl:1]: missing conf section of `clickhouse.table.initializer@span`  in streaming bootstrap.yaml

File: cmd/monitor/streaming/bootstrap.yaml (hunk 4 above)
Comment (erenming) [reply cmd/monitor/streaming/bootstrap.yaml]: ok, already default to 3

**Pull request merged**